### PR TITLE
Build 3.6 and 3.7

### PR DESCRIPTION
**Pull request by paulproteus**
Build 3.6 as well as 3.7

- Create a unified `main.sh` script that can build 3.6 and 3.7.

- Create a unified `python.Dockerfile` that can build 3.6 and 3.7.

- Take command-line parameters, not environment variables, for clarity.

- Adjust `release` workflow accordingly.

- For Python 3.6, add patch from upstream Python 3.7 to include
  `sys.getandroidapilevel()` function, so that Toga can detect our
  Python build as running on Android.

- Create an empty `rubicon.zip` for compatibility with the current
  Android gradle template used by briefcase.

- Move excludes into toplevel `excludes/all` directory, where `all`
  represents the fact that the file is used to filter builds for all
  Python versions.

- Move patches into `patches/3.6` and `patches/3.7`.

- Move script to ignore some stdlib tests into `patches/all` because it
  applies to both 3.6 and 3.7.

## Testing performed

- I have verified that an app starts up! Wowee!!!

- I haven't run the test suite yet. I'm doing that now; still WIP until I get that back.

  - Update: I think manually testing that apps start OK is enough.

### CODE REVIEW

File: main.sh [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    rm -rf ./build/"$VERSION"
+    mkdir -p build/"$VERSION"
+
+    # Move this Python source tarball to python-src.tar.xz. This allows Dockerfile to not
+    # hard-code any specific Python source tarball. For similar reasons, make a copy of
+    # the patches directory into python-src.patches.
+    mkdir -p build-tmp
+    cp -a downloads/Python-"$VERSION"*.tar.xz build-tmp/python-src.tar.xz
+    rm -rf build-tmp/python-src.patches
+    cp -a "$VERSION".patches build-tmp/python-src.patches
```
Comment (paulproteus) [inline main.sh]: Let's do `patches/3.6/` and avoid copying

**Comment by paulproteus** [comment]
I'm marking this "ready for review" even though I haven't gotten through a full test suite run. It's very hard to run the test suite if Android keeps killing my app for "not responding" :( :). I think it's OK as-is.

**Comment by freakboy3742** [comment]
Hrm... odd that this happens for 3.6, but didn't for 3.7...

**Review by freakboy3742 — APPROVED** [review]
👍 All looks good now!

**Pull request merged**